Userland/Libraries/LibWeb/URL/URL.cpp

```diff
@@ -110,153 +110,182 @@ bool URL::can_parse(JS::VM&, String const& url, Optional<String> const& base)
     return true;
 }
 
+// https://url.spec.whatwg.org/#dom-url-href
 WebIDL::ExceptionOr<String> URL::href() const
 {
     auto& vm = realm().vm();
 
-    // return the serialization of this’s URL.
+    // The href getter steps and the toJSON() method steps are to return the serialization of this’s URL.
     return TRY_OR_THROW_OOM(vm, String::from_deprecated_string(m_url.serialize()));
 }
 
+// https://url.spec.whatwg.org/#dom-url-tojson
 WebIDL::ExceptionOr<String> URL::to_json() const
 {
     auto& vm = realm().vm();
 
-    // return the serialization of this’s URL.
+    // The href getter steps and the toJSON() method steps are to return the serialization of this’s URL.
     return TRY_OR_THROW_OOM(vm, String::from_deprecated_string(m_url.serialize()));
 }
 
+// https://url.spec.whatwg.org/#ref-for-dom-url-href②
 WebIDL::ExceptionOr<void> URL::set_href(String const& href)
 {
     auto& vm = realm().vm();
 
     // 1. Let parsedURL be the result of running the basic URL parser on the given value.
     AK::URL parsed_url = href;
+
     // 2. If parsedURL is failure, then throw a TypeError.
     if (!parsed_url.is_valid())
         return WebIDL::SimpleException { WebIDL::SimpleExceptionType::TypeError, "Invalid URL"sv };
+
     // 3. Set this’s URL to parsedURL.
     m_url = move(parsed_url);
+
     // 4. Empty this’s query object’s list.
     m_query->m_list.clear();
+
     // 5. Let query be this’s URL’s query.
     auto& query = m_url.query();
+
     // 6. If query is non-null, then set this’s query object’s list to the result of parsing query.
     if (!query.is_null())
         m_query->m_list = TRY_OR_THROW_OOM(vm, url_decode(query));
     return {};
 }
 
+// https://url.spec.whatwg.org/#dom-url-origin
 WebIDL::ExceptionOr<String> URL::origin() const
 {
     auto& vm = realm().vm();
 
-    // return the serialization of this’s URL’s origin.
+    // The origin getter steps are to return the serialization of this’s URL’s origin. [HTML]
```

Comment:
Post mortem, but I've been leaving out the `[HTML]`  text when copying comments as they're just quick links to the reference section in the spec

Reply:
I've been removing them manually for a while but then went back to just copying verbatim. Needs consensus and a written rule :) (...leaving them out is probably fine)